### PR TITLE
[PR2] Split the TestStand modules into specific modules for the niscope, nifgen and nidmm examples

### PR DESCRIPTION
**Pull request by samchris007**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/measurement-services-python/blob/main/CONTRIBUTING.md).

### What does this Pull Request accomplish?
Split TestStand code modules in the examples by pinmap-specific and driver-specific logic and updated the TestStand sequences accordingly.

The `nidmm_measurement`, `nifgen_standard_function`, and `niscope_acquire_waveform` examples' TestStand code modules are split into driver-specific and pin map-specific code.

### Why should this Pull Request be merged?

Implements https://github.com/ni/measurementlink-python/issues/392 for `nidmm_measurement`, `nifgen_standard_function`, and `niscope_acquire_waveform` examples

### What testing has been done?

- Manually tested by running the sequence examples 
- Ran the existing tests.

### CODE REVIEW

**Review by keer1111 — APPROVED** [review]

File: examples/nidmm_measurement/NIDmmMeasurement.seq [hunk 1]
```diff
@@ -1,13 +1,13 @@
 ﻿<?xml version="1.0" encoding="UTF-8"?>
-<teststandfileheader type='SequenceFile' fileversion='962' productname='TestStand' productversion='2021 SP1 (21.1.0.49154)' compatibleversion='21.0.0.0' buildversion='21.0.0.49156' xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://www.ni.com/TestStand/21.0.0/SequenceFile">
+<teststandfileheader type='SequenceFile' fileversion='962' productname='TestStand' productversion='2021 (21.0.0.49156)' compatibleversion='21.0.0.0' buildversion='21.0.0.49156' xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://www.ni.com/TestStand/21.0.0/SequenceFile">
```
Comment (keer1111) [inline examples/nidmm_measurement/NIDmmMeasurement.seq:2]: version check

File: examples/nidmm_measurement/NIDmmMeasurement.seq
Comment (dixonjoel) [inline examples/nidmm_measurement/NIDmmMeasurement.seq:1]: I get this error when I try to run the sequence. Did you encounter this and how did you get around it?
> Error: Error executing substep 'Post'.
Exception iterating responses: The SESSION_MANAGEMENT_2024Q1 feature is not supported at the current code readiness level.  To enable it, set MEASUREMENTLINK_ENABLE_SESSION_MANAGEMENT_2024Q1 or MEASUREMENTLINK_ALLOW_INCOMPLETE.[Error Code: -17500, Operation Failed. ]

Comment (dixonjoel) [reply examples/nidmm_measurement/NIDmmMeasurement.seq:1]: I was able to overcome this by making a .env file in the root directory of the repo and uncommenting the MEASUREMENTLINK_ENABLE_SESSION_MANAGEMENT_2024Q1 line. I did have to do 'poetry install' on the measurement service again.

**Review by dixonjoel — APPROVED** [review]

File: examples/nidmm_measurement/teststand_pin_map.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+from _helpers import GrpcChannelPoolHelper, PinMapClient, TestStandSupport
+
+
+def update_pin_map(pin_map_path: str, sequence_context: Any) -> str:
+    """Update registered pin map contents.
+
+    Create and register a pin map if a pin map resource for the specified pin map id is not found.
+
+    Args:
+        pin_map_path:
```
Comment (bkeryan) [inline examples/nidmm_measurement/teststand_pin_map.py:13]: https://github.com/ni/measurementlink-python/pull/460#discussion_r1364150727

**Review by bkeryan — CHANGES_REQUESTED** [review]

File: examples/nidmm_measurement/NIDmmMeasurement.seq
Comment (samchris007) [reply examples/nidmm_measurement/NIDmmMeasurement.seq:1]: Yes, that's correct. To run the example measurements with the new session APIs, you need to enable the feature toggle using the `.env` file.  This information has been mentioned [here](https://github.com/ni/measurementlink-python/blob/6d8540ca3bb2b19435252d3b63bde5cdfd86fe18/.env.sample#L4) in our repo

**Review by bkeryan — APPROVED** [review]

**Pull request merged**